### PR TITLE
feat: concept for accounting on ai agent mechs when staking

### PR DESCRIPTION
**Pull request by kupermind**
- Accounting for AI agent mechs specifics when staking services interacting with them.

### CODE REVIEW

File: contracts/staking/ServiceStakingBase.sol [hunk 1]
```diff
@@ -161,13 +163,14 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
 
     /// @dev ServiceStakingBase constructor.
     /// @param _stakingParams Service staking parameters.
+    /// @param _numNonces Number of service multisig nonces depending on implementation.
```
Comment (DavidMinarsch) [inline contracts/staking/ServiceStakingBase.sol:166]: Not sure I follow why we need this

File: contracts/staking/ServiceStakingMechAgentNativeToken.sol [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    /// @dev Gets service multisig nonces.
+    /// @param multisig Service multisig address.
+    /// @return nonces Set of one or more service multisig nonces depending on implementation.
+    function _getMultisigNonces(address multisig) internal view override returns (uint256[] memory nonces) {
+        nonces = new uint256[](numNonces);
+        nonces[0] = IMultisig(multisig).nonce();
+        nonces[1] = IAgentMech(agentMech).getRequestsCount(multisig);
+    }
+
+    /// @dev Checks if the service multisig liveness ratio passes the defined liveness threshold.
```
Comment (DavidMinarsch) [inline contracts/staking/ServiceStakingMechAgentNativeToken.sol:53]: Docstring could explain what exactly liveness means in this context / implementation

File: contracts/staking/ServiceStakingBase.sol (hunk 1 above)
Comment (DavidMinarsch) [reply contracts/staking/ServiceStakingBase.sol:166]: Ok, I see below..

File: contracts/staking/ServiceStakingMechAgentNativeToken.sol [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+// AgentMech interface
+interface IAgentMech {
+    /// @dev Gets the requests count for a specific account.
+    /// @param account Account address.
+    /// @return requestsCount Requests count.
+    function getRequestsCount(address account) external view returns (uint256 requestsCount);
+}
+
+/// @title ServiceStakingMechAgentNativeToken - Smart contract for staking a service with the service interacting wiht
```
Comment (DavidMinarsch) [inline contracts/staking/ServiceStakingMechAgentNativeToken.sol:21]:
```suggestion
/// @title ServiceStakingMechAgentNativeToken - Smart contract for staking a service with the service interacting with
```

**Review by DavidMinarsch — APPROVED** [review]

File: contracts/staking/ServiceStakingBase.sol (hunk 1 above)
Comment (kupermind) [reply contracts/staking/ServiceStakingBase.sol]: Refactored, no need for that now.

**Review by mariapiamo — APPROVED** [review]

**Pull request merged**